## Inference about Proportion

In [1]:
import pandas as pd
from pandas import DataFrame
import numpy as np
from scipy.stats import t, chi2, norm, f
from math import sqrt


<center><img src="Images/Proportion1/5.png" width=750></center>


## Example 1 (Large sample confidence interval for binomial parameter p)
A random sample of 400 computer chips is taken from a large lot of chips and 50 of them are found to be defective. Find a 95% confidence interval for p the proportion of defective chips contained in the lot.

In [9]:
n = 400
T = 50
alpha = 0.05

p_hat = T / n
z = norm.ppf(alpha/2)
z
l = p_hat + z * sqrt(p_hat * (1 - p_hat)/n)
u = p_hat - z * sqrt(p_hat * (1 - p_hat)/n)
print(f"A 95% confidence interval for p is ({l}, {u})")

A 95% confidence interval for p is (0.09259014198914857, 0.15740985801085144)


**Interpretation**: We are 95% confident that the population proportion of defective chips
is between 9.26% and 15.74%.

### 1.3 Inference about a Population Proportion
<center><img src="Images/Proportion1/1.png" width=600></center>
<center><img src="Images/Proportion1/2.png" width=600></center>
The confidence interval estimator of the total number of successes in a large finite population is:
<center><img src="Images/Proportion1/3.png"></center>
<center><img src="Images/Proportion1/4.png" width=600></center>

#### Example 5: Election Day Exit Poll
When an election for political office takes place, the television networks cancel regular programming and instead provide election coverage. When the ballots are counted, the results
are reported. However, for important offices such as president or senator in large states, the
networks actively compete to see which will be the first to predict a winner. This is done
through exit polls, wherein a random sample of voters who exit the polling booth is asked
for whom they voted. From the data, the sample proportion of voters supporting the candidates is computed. A statistical technique is applied to determine whether there is enough
evidence to infer that the leading candidate will garner enough votes to win.
<br>Suppose that in
the exit poll from the state of Florida during the 2000 year elections, the pollsters recorded
only the votes of the two candidates who had any chance of winning, Democrat Albert Gore
(code = 1) and Republican George W. Bush (code = 2). The polls close at 8:00 p.m. Can
the networks conclude from these data that the Republican candidate will win the state?
Should the network announce at 8:01 p.m. that the Republican candidate will win?
<br>
**Solution**: The problem objective is to describe the population of votes in the state. The data are
nominal because the values are “Democrat” (code = 1) and “Republican” (code = 2).
Thus the parameter to be tested is the proportion of votes in the entire state that are
for the Republican candidate. Because we want to determine whether the network can
declare the Republican to be the winner at 8:01 p.m., the alternative hypothesis is <br>
$ H_1: p > .5 $
<br>which makes the null hypothesis<br>
$ H_0: p = .5 $
<br>The test statistic is<br>
$ z = \frac {\hat{p} − p}{\sqrt{\frac{p(1 − p)}{n}}} $

In [ ]:
df = pd.read_csv("Data\\votes.csv")
df.shape
significance = 0.95
p = 0.5
N = df.shape[0]
count_of_2s = df.value_counts()[2]
p_hat = count_of_2s / N
nrm = norm.ppf(0.95)

test_statistics = (p_hat - p) / sqrt(p * (1-p)/N)
print(f"Test Statistics = {test_statistics}")
print(f"Rejection region is: z > {nrm}")
p_value = norm.sf(1.77)
p_value



Because the test statistics lies in the rejection region, we reject the null hypothesis in favor of the alternative hypothesis. There is enough evidence at the 5% significance level that the Republican candidate has won.

#### Example 6: Segmenting the Breakfast Cereal Market (Finding the size of the segment)
In segmenting the breakfast cereal market, a food manufacturer uses health and diet consciousness as the segmentation variable. Four segments are developed:
1. Concerned about eating healthy foods
2. Concerned primarily about weight
3. Concerned about health because of illness
4. Unconcerned

To distinguish between groups, surveys are conducted. On the basis of a questionnaire, people are categorized as belonging to one of these groups. A recent survey asked a random sample of 1,250 American adults (18 and older) to complete the questionnaire. The categories were recorded using the codes. The most recent census reveals that 244,137,873 Americans are 18 and older. Estimate with 95% confidence the number of American adults who are concerned about eating healthy foods.


**Solution:** The problem objective is to describe the population of American adults. The data are nominal. Consequently, the parameter we wish to estimate is the proportion p of American adults who classify themselves as concerned about eating healthy. The confidence interval estimator we need to employ is:
<center><img src="Images/Proportion1/3.png", width=150></center>
from which we will produce the estimate of the size of the market segment.


In [ ]:
df = pd.read_csv("Data\\cereal.csv")
df.shape
significance = 0.975
N = df.shape[0]
count_of_1s = df.value_counts()[1]
p_hat = count_of_1s / N
nrm = norm.ppf(significance)

LCL = p_hat - nrm * sqrt(p_hat * (1-p_hat)/N)
UCL = p_hat + nrm * sqrt(p_hat * (1-p_hat)/N)
print(f"LCL = {LCL}")
print(f"UCL = {UCL}")

us_population = 244137873
left = us_population * LCL
right = us_population * UCL

print(f"Number of American adults who are concerned about eating healthy foods is in the interval:\n({round(left)}, {round(right)})")

## Example (Testing a population proportion)
A set of environmentalists in the United States believe that sport utility vehicles (SUVs) consume excessive amount of
gasoline and are thus “big” polluters of our environment. An environmental agency wants to find what proportion of vehicles on US highways are SUVs. Suppose that a random sample of 500 vehicles collected from highways in various parts of the country shows that 120 out of 500 vehicles are SUVs. Do these data provide sufficient evidence that 25% of
the total vehicles driven in the US are SUVs? Use α= 0.05 level of significance. Find the p-value of the test.

$H0: p = p_0 = 0.25$<br>
$H1: p ≠ 0.25$


In [2]:
alpha = 0.05
n = 500
x = 120
p0 = 0.25

phat = x / n
z = (phat - p0) / sqrt(p0 * (1-p0)/n)
z

-0.5163977794943228

In [4]:
zl = norm.ppf(alpha/2)
zr = norm.ppf(1-alpha/2)
print(f"Acceptance region: ({zl}, {zr})")

Acceptance region: (-1.9599639845400545, 1.959963984540054)


In [6]:
p_value = 2 * norm.cdf(z)
p_value

0.605576616335346

**Interpretation:** Based on the statistical testing procedure with significance level of 0.05, we do not have sufficient evidence (p-value = 0.606) to conclude that the percentage of SUVs on US highways is different from 25%.